storage/range.go

```diff
@@ -769,21 +771,24 @@ func (r *Range) processRaftCommand(idKey cmdIDKey, index uint64, raftCmd proto.I
 
 	args := raftCmd.Cmd.GetValue().(proto.Request)
 	var reply proto.Response
+	var ctx context.Context
 	if cmd != nil {
 		// We initiated this command, so use the caller-supplied reply.
 		reply = cmd.Reply
+		ctx = cmd.ctx
 	} else {
 		// This command originated elsewhere so we must create a new reply buffer.
 		reply = args.CreateReply()
+		ctx = r.context()
 	}
 
-	err := r.applyRaftCommand(index, proto.RaftNodeID(raftCmd.OriginNodeID), args, reply)
+	err := r.applyRaftCommand(ctx, index, proto.RaftNodeID(raftCmd.OriginNodeID), args, reply)
```

Comment:
Not sure I understand. You still want to run `applyRaftCommand` in both cases, but if the command was proposed into Raft locally, you don't need to allocate the reply because you already have one.